### ISSUE TITLE
Hi and code suggestions :)

### ISSUE THREAD

**Issue by jan-wassenberg** [osamu620/JPEGenc#8]
Hello again @osamu620, it is great to see your nice SIMD code :)

We'd be happy to link to your project from the Highway readme.

Some thoughts on the code, in the hope that they are useful:

In quantization.cc and block_coding.cpp maybe we could `HWY_DYNAMIC_DISPATCH` only once because it is an indirect function call, which is relatively expensive. We could move the for(;;) loop inside quantize_fwd?

In dct.cpp, `ZipLower(s32, ..)` is the same as `BitCast(s32, InterleaveLower(s16, ..))` and it could shorten the code a bit.

In color.cpp, it looks like coeffs is 8 int16 constants, and then copied into each 128-bit block.
We can shrink the array to 8 elements if we use `hn::LoadDup128` to load it. And `Broadcast<>` could be used instead of `BroadcastLane<>`: it is a bit cheaper on AVX2 because it is allowed to assume that all 128-bit blocks are equal, which is true here.

I'm also curious whether you see anything we can improve or better document in the Highway library?

**Comment by osamu620** [comment]
@jan-wassenberg Hi Jan, nice to see you again :)  I really appreciate your helpful suggestions. I would be happy for this repo to be referred to in Highway’s document. Highway is great; I believe it brings a unified code base for a variety of SIMD applications.

For the document, it would be nice to have illustrative explanations for the operations to rearrange data in a vector (e.g., interleave, zip, shuffle, etc.)
The example is here (https://www.officedaytime.com/simd512e/)

Because I think it is not easy to understand those operations without any knowledge of existing intrinsics. With the intuitive figures, Highway can be the first choice for people trying their code faster with SIMD technologies.

**Comment by jan-wassenberg** [comment]
:) Thank you for your kind words and feedback. I have added a TODO for such an illustrative guide :)

**Comment by osamu620** [comment]
> :) Thank you for your kind words and feedback. I have added a TODO for such an illustrative guide :)

Thanks! Looking forward to seeing the guide.
